addons/html_editor/static/tests/html_field.test.js

```diff
@@ -679,8 +679,76 @@ test("isDirty should be false when the content is being transformed by the edito
     expect(`.o_form_button_save`).not.toBeVisible();
 });
 
-test.todo("media dialog: upload", async function () {
-    throw new Error("To imp");
+test("link preview in Link Dialog", async () => {
+    Partner._records = [
+        {
+            id: 1,
+            txt: "<p class='test_target'><a href='/test'>This website</a></p>",
+        },
+    ];
+    await mountView({
+        type: "form",
+        resId: 1,
+        resModel: "partner",
+        arch: `
+            <form>
+                <field name="txt" widget="html"/>
+            </form>`,
+    });
+
+    expect(".test_target a").toHaveText("This website");
+
+    // Open the popover option to edit the link
+    let anchorNode = queryOne(".test_target a");
+    setSelection({ anchorNode, anchorOffset: 0 });
+    // Click on the edit link icon
+    await contains("a.o_we_edit_link").click();
+    expect(".o-we-linkpopover input.o_we_label_link").toHaveValue("This website", {
+        message: "The label input field should match the link's content",
+    });
+    expect(".o-we-linkpopover a#link-preview").toHaveText("This website", {
+        message: "Link label in preview should match label input field",
+    });
+
+    await contains(".o-we-linkpopover input.o_we_label_link").edit("Bad new label");
+    expect(".o-we-linkpopover input.o_we_label_link").toHaveValue("Bad new label", {
+        message: "The label input field should match the link's content",
+    });
+    expect(".o-we-linkpopover a#link-preview").toHaveText("Bad new label", {
+        message: "Link label in preview should match label input field",
+    });
+    // Move selection outside to discard
+    anchorNode = queryOne(".test_target");
+    setSelection({ anchorNode, anchorOffset: 0 });
+    await animationFrame();
+    expect(".o-we-linkpopover").toHaveCount(0);
+    expect(".test_target a").toHaveText("This website");
+
+    anchorNode = queryOne(".test_target a");
+    // Select link label to open the floating toolbar.
+    setSelection({ anchorNode, anchorOffset: 0 });
+    await animationFrame();
+    // Click on the edit link icon
+    await contains("a.o_we_edit_link").click();
+    expect(".o-we-linkpopover input.o_we_label_link").toHaveValue("This website", {
+        message: "The label input field should match the link's content",
+    });
+    expect(".o-we-linkpopover a#link-preview").toHaveText("This website", {
+        message: "Link label in preview should match label input field",
+    });
+
+    // Open the popover option to edit the link
+    await contains(".o-we-linkpopover input.o_we_label_link").edit("New label");
+    expect(".o-we-linkpopover a#link-preview").toHaveText("New label", {
+        message: "Preview should be updated on label input field change",
+    });
+
+    // Click "Save".
+    await contains(".o-we-linkpopover .o_we_apply_link").click();
+    expect(".o-we-linkpopover").toHaveCount(0);
```

Comment:
The linkpopover should still be 1 but not in editing mode because the selection is still inside the link, for example
```suggestion
    expect(".o-we-linkpopover").toHaveCount(1);
    expect(".o_we_copy_link").toHaveCount(1);
    expect(".o_we_edit_link").toHaveCount(1);
    expect(".o_we_remove_link").toHaveCount(1);
```